pkg/envinfo/envinfo_test.go

```diff
@@ -0,0 +1,247 @@
+package envinfo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/openshift/odo/pkg/testingutil/filesystem"
+
+	"github.com/openshift/odo/pkg/util"
+)
+
+func TestSetEnvInfo(t *testing.T) {
+
+	tempEnvFile, err := ioutil.TempFile("", "odoenvinfo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tempEnvFile.Close()
+	os.Setenv(envInfoEnvName, tempEnvFile.Name())
+	testURL := ConfigURL{Name: "testURL", ClusterHost: "1.2.3.4.nip.io", TLSSecret: "testTLSSecret"}
+
+	tests := []struct {
+		name            string
+		parameter       string
+		value           interface{}
+		existingEnvInfo EnvInfo
```

Comment:
please add a parameter for checking errors too

Reply:
updated